### DISCUSSION TITLE
Missing borders on gnome apps

### DISCUSSION THREAD

**Discussion by delacrucen** [regolith-linux/regolith-desktop#748] [Q&A]
Hi, pretty much the title.
Terminal apps have borders but not gnome apps like files or everything else.
Is there a way to config regolith/tmux to give borders to these kind of apps?

Files app focused:
![2022-10-14_20-15](https://user-images.githubusercontent.com/58452509/195956588-a178c3f8-fc9c-4238-822c-07ffe5b645b1.png)

Terminal focused:
![2022-10-14_20-14](https://user-images.githubusercontent.com/58452509/195956627-588be18d-c1af-4c6a-836b-ccba55040cbf.png)

Found information in some reddit posts but I assume it goes somewhere else after trying on .config/regolith2/Xresources
The suggested line was: for_window [class="appname"] pixel 0

Thanks.

**Comment by jtsao22** [comment]
That is correct except you want it in the file:
$HOME/.config/regolith/i3/config

For instance for firefox, I use:
for_window [class="firefox"] border pixel 1

To find the appname, you can type "xprop" in the terminal and then select the window of the application you want to use, then use one the strings in the "WM_CLASS(STRING)"

After modifying the i3 config, refresh your i3 session (I changed my hotkey, but you can look it up using the keyboard short cut manager (Mod + Shift + ?)), and you should be good.

**Reply by ghost** [reply]
Thanks for your answer.

I've tried that in `/etc/i3/config` since I'm using regolith2, but I still see coloured borders only on terminal windows. Any idea if this config might get overruled by another config?

**Reply by jtsao22** [reply]
Yeah so the precedence order is:
If $HOME/.config/regolith/i3/config exists, this will be used.
If it doesn't exist, then /etc/i3/config will be used.

**Reply by ghost** [reply]
For me on Regolith2 it was actually yet another place where I needed to config it: `/usr/share/regolith/i3/config.d/40_default-style`

**Reply by delacrucen** [reply]
That worked for me! Thanks @JakobRieckhGraf.
Also found a way to put border on any app, like settings for example.
`for_window [class=".*"] border pixel 1`

**Comment by kgilmer** [comment]
@JakobRieckhGraf , while changing the config file in `/usr` works, the next time you upgrade , it will likely overwrite your changes.  There is an approach to override some configuration such that it won't be clobbered by future updates: https://regolith-desktop.com/docs/howtos/customize-i3-configuration/

Also, FYI, in the upcoming Regolith 2.2, this border setting can be achieved by setting the Xresource key `i3-wm.window.border.size`.  No i3 config changed required.

**Comment by adiSuper94** [comment]
@kgilmer I know this is an old thread, but I facing an issue with border highlight. This isn't even specific to gnome-apps.
When I have multiple windows open in a single workspace the highlight work perfectly. Even when I do the mod+v/g, the next window highlight also works perfectly.

The `i3-wm.window.border.size` doesn't seem to be working for me.

But when I have only one window open in a workspace, the border highlight is missing. 
I know there is little need border highlight when there is only one window in a workspace, but I really liked my i3 setup where I could visually see where the next window would open.
